pkg/cmd/build/build.go

```diff
@@ -212,67 +212,172 @@ func translateDockerErr(err error) error {
+	}
+	if len(cmdBuildOptions.CacheFrom) > 0 {
+		manifestBuildOptions.CacheFrom = cmdBuildOptions.CacheFrom
+	}
+	if cmdBuildOptions.ExportCache != manifestBuildOptions.ExportCache {
+		manifestBuildOptions.ExportCache = cmdBuildOptions.ExportCache
+	}
 
+	// override output mode from cmdBuildOptions
+	if cmdBuildOptions.OutputMode != "" {
+		manifestBuildOptions.OutputMode = setOutputMode(cmdBuildOptions.OutputMode)
+	} else if manifestBuildOptions.OutputMode == "" {
+		manifestBuildOptions.OutputMode = setOutputMode(oktetoLog.GetOutputFormat())
 	}
 
-	file := b.Dockerfile
-	if !filepath.IsAbs(b.Dockerfile) && !model.FileExistsAndNotDir(file) {
-		file = filepath.Join(b.Context, b.Dockerfile)
+	if cmdBuildOptions.Tag != "" {
+		manifestBuildOptions.Tag = cmdBuildOptions.Tag
 	}
-	opts := &BuildOptions{
-		CacheFrom:   b.CacheFrom,
-		Target:      b.Target,
-		Path:        b.Context,
-		Tag:         b.Image,
-		File:        file,
-		BuildArgs:   args,
-		ExportCache: b.ExportCache,
+
+	// override Tag if global registry
+	if okteto.Context().IsOkteto && cmdBuildOptions.BuildToGlobal {
+		manifestBuildOptions.Tag = registry.ReplaceTargetRepository(manifestBuildOptions.Tag, okteto.GlobalRegistry, manifestBuildOptions.Namespace)
 	}
 
-	outputMode := oktetoLog.GetOutputFormat()
-	if o != nil && o.OutputMode != "" {
-		outputMode = o.OutputMode
+	return manifestBuildOptions
+}
+
+// OptsFromManifest returns the parsed options for the build from the manifest
+func OptsFromManifest(service string, manifestBuildInfo *model.BuildInfo, cmdBuildOptions *BuildOptions) *BuildOptions {
+	manifestBuildOptions := getBuildOptionsFromManifest(service, manifestBuildInfo)
+	return overrideManifestBuildOptions(manifestBuildOptions, cmdBuildOptions)
+}
+
+// optimizedGlobalBuild returns if build should be optimized because the image is already at the global registry
+func (opts *BuildOptions) optimizedGlobalBuild() (string, bool, error) {
+	oktetoLog.Debugf("Applying global build optimization for image %s", opts.Tag)
+
+	globalReference := opts.Tag
+	if registry.IsDevRegistry(opts.Tag) || registry.IsExtendedOktetoRegistry(opts.Tag) {
+		globalReference = registry.ReplaceTargetRepository(opts.Tag, okteto.GlobalRegistry, okteto.DefaultGlobalNamespace)
+
 	}
-	opts.OutputMode = setOutputMode(outputMode)
+	tagWithDigest, err := registry.GetImageTagWithDigest(globalReference)
+	if err != nil {
+		return opts.Tag, false, err
+	}
+	oktetoLog.Debugf("skipping build: image %s is already built", opts.Tag)
+	opts.Tag = globalReference
+	return tagWithDigest, true, nil
+}
 
-	return opts
+// checkImageAtRegistry returns if build should be optimized because the image is already at the dev registry
+func (opts *BuildOptions) checkImageAtRegistry() (string, bool, error) {
+	oktetoLog.Debugf("Checking registry for image %s", opts.Tag)
+	tagWithDigest, err := registry.GetImageTagWithDigest(opts.Tag)
+	if err != nil {
+		return opts.Tag, false, err
```

Comment:
This is used on `SkipBuild` function which says `returns if build has to be skipped and the tag with digest if found at registry` but here (and in `optimizedGlobalBuild`), we are returning the tag even when there is an error checking in the registry (so we are not sure the tag is in the registry), it is on purpose? If so, I think it is a bit misleading

Reply:
Yes, the idea for this two functions is to return if the build should be skipped (bool) and the tag with digest (string) so its already there and we dont call again the registry. In case of error i guess the best would be to return empty string, false, and the error. Also, i see that maybe i can remove the bool value and just return (string, error) where if string is not empty, the digest has been found so we can work with that.